pkg/web/transactions.go

```diff
@@ -455,10 +457,24 @@ func (s *Server) SendEnvelopeForTransaction(c *gin.Context) {
 		return
 	}
 
-	// TODO: we need to get the incoming envelope in order for this to work!
+	// Retrieve the secure envelope model for the incoming envelope
+	if incoming, err = s.store.LatestSecureEnvelope(ctx, transaction.ID, models.DirectionIncoming); err != nil {
+		c.Error(fmt.Errorf("could not retrieve incoming secure envelope: %w", err))
+		c.JSON(http.StatusInternalServerError, api.Error("could not return incoming response from counterparty"))
+		return
+	}
+
+	// Decrypt the incoming secure envelope
+	// TODO: why are we decrypting the incoming secure envelope again?
```

Comment:
Are you asking if the secure envelope should be decrypted?

Reply:
No, it's just that it's the second time the decryption is happening; the first time when it came into the node, then was saved to the database (encrypted) then we decrypt it again after loading it from the database; it's wasted extra work, but I'd have to be able to access the incoming message in memory somehow to not do the second decryption.

I think this means creating a better sending mechanism that wraps the various functionality required.

Reply:
That makes sense. I saw this first and then saw this comment elsewhere and wanted to make sure I understood the thought process.